### PR TITLE
ETSWORK-10058 - Add support for push notifications via `@capacitor/push-notifications` and Firebase.

### PR DESCRIPTION
**Pull request by coreyjv**
The app was configured per the [Capacitor v3 Push Notifications guide](https://capacitorjs.com/docs/v3/guides/push-notifications-firebase#add-initialization-code). The implementation uses a provider pattern similar to the other providers that exist already. A provider pattern was used because:

1. We need to have an authentication token and API host so it must be a child of `AuthenticationProvider`
2. From a UX perspective the push notification popup needs to be displayed after a space is selected
3. Putting it outside the router maintains the required component order for Ionic to work properly. This is why `history` is created outside and passed in to both this and `IonReactRouter`. History is used for deep linking
4. We may expose a hook in the future to allow the display of the current status, test a notification and/or re-registration on app settings screens.

This pattern will also call the Workgrid device registration endpoint every time the app is restarted. This was intentional to ensure we always have the latest registration token from the device.

Finally this PR does not implement Workgrid specific deep linking support. This will come in a future update. What this PR does however is show how we can push state on to history and render that as shown on the Home screen.

### CODE REVIEW

File: android/app/google-services.json [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    "firebase_url": "https://workgrid-208419.firebaseio.com",
+    "project_id": "workgrid-208419",
+    "storage_bucket": "workgrid-208419.appspot.com"
+  },
+  "client": [
+    {
+      "client_info": {
+        "mobilesdk_app_id": "1:495897271921:android:946b76083d86351bef85d0",
+        "android_client_info": {
+          "package_name": "com.libertymutual.workgrid.iconicpoc"
```
Comment (TuckerWhitehouse) [inline android/app/google-services.json:13]: We could let this typo die... :)

File: ios/App/App/AppDelegate.swift [hunk 2]
```diff
@@ -60,13 +62,16 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+                NotificationCenter.default.post(name: .capacitorDidRegisterForRemoteNotifications, object: token)
+            }
+          })
     }
-
-    func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
-        NotificationCenter.default.post(name: .capacitorDidFailToRegisterForRemoteNotifications, object: error)
-    }
-
+
```
Comment (TuckerWhitehouse) [inline ios/App/App/AppDelegate.swift:74]: I'm sure this is copy/paste from the docs, but seems like removing this handler would mean we wouldn't be handling this scenario, but I don't really understand the difference between this and the callback in `Messaging.message().token()`

File: src/components/Home.tsx [hunk 3]
```diff
@@ -27,12 +28,21 @@ const Home: React.FC = () => {
   const { signOut } = useAuth()
   const { currentSpaceId, spaces, onSpaceChange } = useSpaces()
   const { t } = useTranslation('authentication')
+  const location = useLocation<{ notification: { title: string; body: string; id: string } }>()
 
+  console.log('location.state', location.state)
```
Comment (TuckerWhitehouse) [inline src/components/Home.tsx:33]:
```suggestion
```

File: src/services/history.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+import { createBrowserHistory } from 'history'
+
+// TODO This will likely have to change to `createHashHistory` when we pull in Electron support as that requires a HashRouter
```
Comment (TuckerWhitehouse) [inline src/services/history.ts:3]: I wonder if we can do this conditionally so clients that can properly support regular urls get them, and electron (where you can't see the url) uses hash or in memory.

**Review by TuckerWhitehouse — APPROVED** [review]

File: ios/App/App/AppDelegate.swift (hunk 2 above)
Comment (coreyjv) [reply ios/App/App/AppDelegate.swift:74]: This configuration is straight from the documentation. I'm just guessing here but with Firebase I suspect it cannot fail at the platform level as there's an error branch in the success handler that is called out in the Capacitor guide.

File: src/services/history.ts (hunk 4 above)
Comment (coreyjv) [reply src/services/history.ts:3]: It seems plausible and we can tackle that when/if we integrate `@capacitor-community/electron` back into the app.

**Review by sjjharrington — APPROVED** [review]

File: android/app/google-services.json (hunk 1 above)
Comment (coreyjv) [reply android/app/google-services.json:13]: Ya so I removed the old apps from the project and this still exists in the `google-services.json` so I don't think it's going anywhere anytime soon.

**Review by TuckerWhitehouse — APPROVED** [review]

**Pull request merged**